src/Components/Forms/src/EditContextProperties.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.AspNetCore.Components.Forms
+{
+    /// <summary>
+    /// Holds arbitrary key/value pairs associated with an <see cref="EditContext"/>.
+    /// This can be used to track additional metadata for application-specific purposes.
+    /// </summary>
+    public sealed class EditContextProperties
+    {
+        // We don't want to expose any way of enumerating the underlying dictionary, because that would
+        // prevent its usage to store private information. So we only expose an indexer and TryGetValue.
+        private Dictionary<object, object>? _contents;
+
+        /// <summary>
+        /// Gets or sets a value in the collection.
+        /// </summary>
+        /// <param name="key">The key under which the value is stored.</param>
+        /// <returns>The stored value.</returns>
+        public object this[object key]
```

Comment:
How do you feel about making this a `Set` method? It has slightly less behavior.

Reply:
I'm not sure what you mean by having less behavior. Can you ping me on IM when you're available to clarify?

Overall I don't really mind what the API looks like here since it's so obscure - any reasonable option is fine with me.

Reply:
It does not seem like we need the getter at all. If you think it's necessary, we should update the KeyNotFoundException to use the same error message as the dictionary's exception